lib/mix/lib/mix/dep.ex

```diff
@@ -167,10 +167,10 @@ defmodule Mix.Dep do
     do: "lock mismatch: the dependency is out of date (run \"mix deps.get\" to fetch locked version)"
```

Comment:
Maybe if current version is < than mix.lock with should the out of date, but if not we should the "does not match version in mix.lock".. WDYT?